### PR TITLE
Added getShorURLSynchronous

### PR DESCRIPTION
**Pull request by qinweigong**
@aaustin 

Also, 1) excluded close from persist queue, 2) stop operation if ![self hasUser]

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BNCServerInterface.m [hunk 1]
```diff
@@ -130,35 +116,75 @@ + (NSString *)encodePostToUniversalString:(NSDictionary *)params needSource:(BOO
...
+    BNCServerResponse *serverResponse = nil;
+    if (!error) {
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+        NSNumber *statusCode = [NSNumber numberWithLong:[httpResponse statusCode]];
+        
+        serverResponse = [[BNCServerResponse alloc] initWithTag:requestTag andStatusCode:statusCode];
+        serverResponse.linkData = linkData;
+        
+        if (POSTReply != nil) {
+            NSError *convError;
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: Unused

File: Branch-SDK/Branch-SDK/Branch.m [hunk 2]
```diff
@@ -878,6 +915,59 @@ - (void)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andType:(Bra
...
+    [post setupType:type];
+    [post setupTags:tags];
+    [post setupChannel:channel];
+    [post setupFeature:feature];
+    [post setupStage:stage];
+    [post setupAlias:alias];
+    
+    [post setObject:[NSString stringWithFormat:@"ios%@", SDK_VERSION] forKey:@"sdk"];
+    
+    NSString *args = params ? params : @"{ \"source\":\"ios\" }";
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: You can use `params ?: @"...";` here, just FYI.

Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: Haha, for better readability :-)

File: Branch-SDK/Branch-SDK/BNCServerInterface.m (hunk 1 above)
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: Which line is unused?

File: Branch-SDK/Branch-SDK/Branch.m (hunk 2 above)
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: Sure, either way is fine. Just wasn't sure if you were familiar with the syntax.

File: Branch-SDK/Branch-SDK/BNCServerInterface.m (hunk 1 above)
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: The error here is passed into that serialization, but is never checked.

File: Branch-SDK/Branch-SDK/Branch.m (hunk 2 above)
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: No, I didn't know about it until now :-)

File: Branch-SDK/Branch-SDK/BNCServerInterface.m (hunk 1 above)
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: Ah, i see!

**Comment by aaustin** [comment]
:+1: looks great

**Pull request merged**